src/DelegateDecompiler/ReferencedQueryableDecompilerVisitor.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace DelegateDecompiler
+{
+    public class ReferencedQueryableExpressionVisitor
+        : ExpressionVisitor
+    {
+        public static Expression Decompile(Expression expression)
+        {
+            return new ReferencedQueryableExpressionVisitor().Visit(expression);
+        }
+
+        public override Expression Visit(Expression node)
+        {
+            var result = base.Visit(node);
+            return result;
+        }
+
+        protected override Expression VisitMember(MemberExpression node)
+        {
+            if (typeof(IQueryable).IsAssignableFrom(node.Type))
+            {
+                if (Configuration.Instance.ShouldDecompile(node.Member))
```

Comment:
We forcibly check against the configuration (thus the [Decompile] requirements on the Tests) to avoid conflicts and stack-overflows against System.Linq.Queryable extensions or third-party code